### PR TITLE
Remove calls to `onConsecutiveCalls()`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

cc @OskarStark as we did a lot of stuff on tests migration last year

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Tests/Messenger/DoctrineOpenTransactionLoggerMiddlewareTest.php [hunk 1]
```diff
@@ -52,8 +52,11 @@ public function testMiddlewareWrapsInTransactionAndFlushes()
     {
         $this->connection->expects($this->exactly(1))
             ->method('isTransactionActive')
-            ->will($this->onConsecutiveCalls(true, true, false))
-        ;
+            ->willReturnCallback(function () {
+                static $expected = [true, true, false];
+
+                return array_shift($expected);
+            });
```
Comment (derrabus) [inline src/Symfony/Bridge/Doctrine/Tests/Messenger/DoctrineOpenTransactionLoggerMiddlewareTest.php]:
```suggestion
            ->willReturn(true, true, false);
```

Comment (alexandre-daubois) [reply src/Symfony/Bridge/Doctrine/Tests/Messenger/DoctrineOpenTransactionLoggerMiddlewareTest.php]: I was not aware of this! It will greatly simplify the code. I'll update, thank you!

Comment (alexandre-daubois) [reply src/Symfony/Bridge/Doctrine/Tests/Messenger/DoctrineOpenTransactionLoggerMiddlewareTest.php]: Cuts the PR size in half, way to go 🙌

File: src/Symfony/Component/Lock/Tests/LockTest.php [hunk 2]
```diff
@@ -313,7 +313,8 @@ public function testReleaseThrowsExceptionWhenDeletionFail()
         $store
             ->expects($this->never())
             ->method('exists')
-            ->with($key);
+            ->with($key)
+            ->willReturn(true);
```
Comment (xabbuh) [inline src/Symfony/Component/Lock/Tests/LockTest.php:317]: Is this change necessary?

Comment (alexandre-daubois) [reply src/Symfony/Component/Lock/Tests/LockTest.php:317]: Yes, otherwise the code crashes: `Lock::__destruct()` calls `isAcquired()`, which ultimately depends on the return value of the `exists` method of the store. Without giving a return value, `null` is used, which makes `isAcquired()` return null. This throws a TypeError, as `isAcquired()` has its return type set with `bool`.

File: src/Symfony/Component/Mailer/Tests/Transport/RoundRobinTransportTest.php [hunk 3]
```diff
@@ -120,10 +120,18 @@ public function testSendOneDeadAndRecoveryWithinRetryPeriod()
...
-        $t2->expects($this->exactly(2))
+        $t2->expects($matcher)
             ->method('send')
-            ->willReturnOnConsecutiveCalls($this->throwException(new TransportException()));
+            ->willReturnCallback(function () use ($matcher) {
+                if (1 === $matcher->getInvocationCount()) {
+                    throw new TransportException();
+                }
+
+                return null;
```
Comment (xabbuh) [inline src/Symfony/Component/Mailer/Tests/Transport/RoundRobinTransportTest.php:133]: ~~`send()` expects an `Envelope` to be returned~~

File: src/Symfony/Component/Validator/Tests/Mapping/Loader/PropertyInfoLoaderTest.php [hunk 4]
```diff
@@ -222,11 +224,12 @@ public function testClassNoAutoMapping()
...
             ->willReturn(['string', 'autoMappingExplicitlyEnabled'])
         ;
+
         $propertyInfoStub
             ->method('getTypes')
-            ->willReturnOnConsecutiveCalls(
+            ->willReturn(
                 [new Type(Type::BUILTIN_TYPE_STRING)],
-                [new Type(Type::BUILTIN_TYPE_BOOL)]
+                [new Type(Type::BUILTIN_TYPE_STRING)]
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Mapping/Loader/PropertyInfoLoaderTest.php]: Why this change?

File: src/Symfony/Component/Mailer/Tests/Transport/FailoverTransportTest.php [hunk 5]
```diff
@@ -85,16 +85,30 @@ public function testSendOneDead()
...
+
+        $t1Matcher = $this->any();
+        $t1->expects($t1Matcher)
+            ->method('send')
+            ->willReturnCallback(function () use ($t1Matcher) {
+                if (1 === $t1Matcher->getInvocationCount()) {
+                    throw new TransportException();
+                }
+
+                return null;
```
Comment (xabbuh) [inline src/Symfony/Component/Mailer/Tests/Transport/FailoverTransportTest.php:97]: `send()` expects an `Envelope` to be returned (same everywhere below)

Comment (alexandre-daubois) [reply src/Symfony/Component/Mailer/Tests/Transport/FailoverTransportTest.php:97]: The current behavior is that it returns `null` after the first call (because `willReturnOnConsecutiveCalls` is only call with one argument, all other calls will return null). Also, `TransportInterface` [allows to return `null`](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Mailer/Transport/TransportInterface.php) when calling `send`. Should we really make this change? You may confuse with `TransportInterface` from Messenger?

File: src/Symfony/Component/Validator/Tests/Mapping/Loader/PropertyInfoLoaderTest.php (hunk 4 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Validator/Tests/Mapping/Loader/PropertyInfoLoaderTest.php]: It should not have changed, my bad!

File: src/Symfony/Component/Mailer/Tests/Transport/FailoverTransportTest.php (hunk 5 above)
Comment (xabbuh) [reply src/Symfony/Component/Mailer/Tests/Transport/FailoverTransportTest.php:97]: My bad, I looked at the wrong interface. 🙈

**Review by xabbuh — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alexandre-daubois.
